### PR TITLE
Simnet and testnet dockers

### PR DESCRIPTION
**Pull request by someanon**
Should be merged after https://github.com/bitlum/connector/pull/4

### CODE REVIEW

File: docker/simnet/bitcoin-cash/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
+FROM ubuntu:18.04 AS builder
+
+ARG BITCOIN_CASH_VERSION
+
+RUN apt-get update && apt-get install -y \
```
Comment (AndrewSamokhvalov) [inline docker/simnet/bitcoin-cash/Dockerfile]: Why do we need that?

File: docker/simnet/bitcoin-cash/bitcoin.simnet.secondary.conf [hunk 2]
```diff
@@ -0,0 +1,20 @@
+regtest=1
```
Comment (AndrewSamokhvalov) [inline docker/simnet/bitcoin-cash/bitcoin.simnet.secondary.conf]: would be better to use predefined configs with explanation of every line

File: docker/simnet/bitcoin-lightning/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+# Copying required binaries from builder stage.
+COPY --from=builder /go/bin/lnd /go/bin/lncli /usr/local/bin/
+
+# Default config used to initalize datadir volume at first or
+# cleaned deploy.
+COPY bitcoin-lightning.simnet.$ROLE.conf /root/default/lnd.conf
+
+# Entrypoint script used to init datadir if required and for
+# starting bitcoin daemon.
```
Comment (AndrewSamokhvalov) [inline docker/simnet/bitcoin-lightning/Dockerfile]: bitcoin daemon?

File: docker/simnet/bitcoin-lightning/bitcoin-lightning.simnet.primary.conf [hunk 4]
```diff
@@ -0,0 +1,23 @@
+[Application Options]
+
+maxpendingchannels=10
```
Comment (AndrewSamokhvalov) [inline docker/simnet/bitcoin-lightning/bitcoin-lightning.simnet.primary.conf]: On lnd repo there is config with every config line explained, I believe we should need to put those comments there.

File: docker/simnet/connector/connector.simnet.primary.conf [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+dash.user=user
+dash.password=password
+dash.feeperunit=350
+
+[Ethereum]
+ethereum.minconfirmations=1
+ethereum.syncdelay=5
+ethereum.host=ethereum.simnet.primary
+ethereum.port=11332
+ethereum.password=
```
Comment (AndrewSamokhvalov) [inline docker/simnet/connector/connector.simnet.primary.conf]: why password is empty?

File: docker/simnet/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,410 @@
...
+  ######################################################################
+
+  blocks-generator.simnet:
+    container_name: blocks-generator.simnet
+    image: blocks-generator.simnet
+    image: blocks-generator.simnet
+    build:
+      context: ./blocks-generator
+      args:
+        - BITCOIN_VERSION
```
Comment (AndrewSamokhvalov) [inline docker/simnet/docker-compose.yml]: why do we need it here?

File: docker/simnet/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,410 @@
...
+  #
+  # This containers are used both by primary and secondary containers
+  # and should be started only once.
+  #
+  ######################################################################
+
+  blocks-generator.simnet:
+    container_name: blocks-generator.simnet
+    image: blocks-generator.simnet
+    image: blocks-generator.simnet
```
Comment (AndrewSamokhvalov) [inline docker/simnet/docker-compose.yml]: why do we have the same lines?

File: docker/simnet/docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,410 @@
...
+    logging:
+      driver: journald
+    restart: always
+    depends_on:
+      - bitcoin.simnet.primary
+      - bitcoin-cash.simnet.primary
+      - dash.simnet.primary
+      - litecoin.simnet.primary
+
+  ethereum-bootnode.simnet:
```
Comment (AndrewSamokhvalov) [inline docker/simnet/docker-compose.yml]: what is that? could we place comment on top of the containers?

File: docker/simnet/docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,410 @@
...
+    container_name: ethereum-bootnode.simnet
+    image: ethereum-bootnode.simnet
+    build:
+      context: ./ethereum-bootnode
+      args:
+        - ETHEREUM_REVISION
+    environment:
+      - EXTERNAL_IP
```
Comment (AndrewSamokhvalov) [inline docker/simnet/docker-compose.yml]: Why external ip is needed on simnet network?

File: docker/simnet/docker-compose.yml [hunk 10]
```diff
@@ -0,0 +1,410 @@
...
+        ipv4_address: 172.100.0.101
+    volumes:
+      - /connector/bitcoin.simnet.primary:/root/.bitcoin:ro
+      - /connector/bitcoin-lightning.simnet.primary:/root/.lnd
+    logging:
+      driver: journald
+    restart: always
+    depends_on:
+      - bitcoin.simnet.primary
+      - blocks-generator.simnet
```
Comment (AndrewSamokhvalov) [inline docker/simnet/docker-compose.yml]: why it depends on blockchain generator?

File: docker/simnet/docker-compose.yml [hunk 11]
```diff
@@ -0,0 +1,410 @@
...
+      context: ./dash
+      args:
+        - DASH_VERSION
+        - ROLE=primary
+    environment:
+      - EXTERNAL_IP
+    networks:
+      connector.simnet:
+    volumes:
+      - /connector/dash.simnet.primary:/root/.dashcore
```
Comment (AndrewSamokhvalov) [inline docker/simnet/docker-compose.yml]: why .dashcore rather than .dash?

File: docker/simnet/docker-compose.yml [hunk 12]
```diff
@@ -0,0 +1,410 @@
...
+    container_name: ethereum.simnet.primary
+    image: ethereum.simnet.primary
+    build:
+      context: ./ethereum
+      args:
+        - ETHEREUM_REVISION
+        - ROLE=primary
+    environment:
+      - EXTERNAL_IP
+      # MINE to mine or not, should be 1 or 0.
```
Comment (AndrewSamokhvalov) [inline docker/simnet/docker-compose.yml]: Isn't that defined by primary vs secondary node?

File: docker/simnet/docker-compose.yml [hunk 13]
```diff
@@ -0,0 +1,410 @@
...
+    build:
+      context: ./connector
+      args:
+        - ROLE=secondary
+    networks:
+      connector.simnet:
+        ipv4_address: 172.100.0.200
+    volumes:
+      - /connector/connector.simnet.secondary:/root/.connector
+      - /connector/bitcoin-lightning.simnet.secondary:/root/.lnd/:ro
```
Comment (AndrewSamokhvalov) [inline docker/simnet/docker-compose.yml]: why connector is needed the lnd volume? (I know it), but somebody else will not

File: docker/simnet/ethereum-bootnode/Dockerfile [hunk 14]
```diff
@@ -0,0 +1,26 @@
+FROM golang:1.9.6 AS builder
+
+ARG ETHEREUM_REVISION
+
+WORKDIR /ethereum
+
+RUN curl -L https://github.com/bitlum/go-ethereum/archive/$ETHEREUM_REVISION.tar.gz \
```
Comment (AndrewSamokhvalov) [inline docker/simnet/ethereum-bootnode/Dockerfile]: Could we please add comment everywhere on top such commands?

File: docker/testnet/connector/connector.testnet.conf [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+bitcoin.user=user
+bitcoin.password=password
+bitcoin.feeperunit=87
+
+[Bitcoincash]
+bitcoincash.minconfirmations=1
+bitcoincash.syncdelay=5
+bitcoincash.host=bitcoin-cash.testnet
+bitcoincash.port=9332
+bitcoincash.user=user
```
Comment (AndrewSamokhvalov) [inline docker/testnet/connector/connector.testnet.conf]: Are you okay with passwords being exposed to the public?

File: docker/simnet/docker-compose.yml (hunk 11 above)
Comment (someanon) [reply docker/simnet/docker-compose.yml]: It is default datadir for dash.

**Comment by someanon** [comment]
Obselete branch

**Pull request closed without merging**